octopoes/octopoes/models/__init__.py

```diff
@@ -11,8 +11,8 @@
 class Reference(str):
     @classmethod
     def parse(cls, ref_str: str) -> tuple[str, str]:
-        object_type, *natural_key_parts = ref_str.split("|")
-        return object_type, "|".join(natural_key_parts)
+        object_type, *natural_key_parts = ref_str.split("|", 1)
+        return object_type, natural_key_parts[0] if natural_key_parts else ""
```

Comment:
I read the code incorrectly, it indeed doesn't change the functionality. Given that a reference always contains the class and the natural key, we can also simplify this to

```suggestion
        return ref_str.split("|", 1)
```